File: app/src/main/java/com/chess/cryptobot/view/adapter/PairsAdapter.java
package com.chess.cryptobot.view.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.chess.cryptobot.R;
import com.chess.cryptobot.content.ContextHolder;
import com.chess.cryptobot.model.Pair;

import java.util.Locale;

public class PairsAdapter extends RecyclerViewAdapter<PairsAdapter.PairsViewHolder> {

    public PairsAdapter(ContextHolder pairsHolder) {
        super(pairsHolder);
    }

    @NonNull
    @Override
    public PairsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater layoutInflater = LayoutInflater.from(context);

        return new PairsViewHolder(layoutInflater.inflate(R.layout.trading_pair_line, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull PairsViewHolder holder, int position) {
        Pair pair = (Pair) getItemByPosition(position);

        holder.pairNameView.setText(pair.getName());
        holder.livecoinBidView.setText(String.format(Locale.getDefault(), "%.8f", pair.getLivecoinBid()));
        holder.livecoinBidQuantityView.setText(String.format(Locale.getDefault(), "%.8f", pair.getLivecoinBidQuantity()));
        holder.livecoinAskView.setText(String.format(Locale.getDefault(), "%.8f", pair.getLivecoinAsk()));
        holder.livecoinAskQuantityView.setText(String.format(Locale.getDefault(), "%.8f", pair.getLivecoinAskQuantity()));
        holder.livecoinVolumeView.setText(String.format(Locale.getDefault(), "%.10f", pair.getLivecoinVolume()));
        holder.bittrextBidView.setText(String.format(Locale.getDefault(), "%.8f", pair.getBittrexBid()));
        holder.bittrextBidQuantityView.setText(String.format(Locale.getDefault(), "%.8f", pair.getBittrexBidQuantity()));
        holder.bittrexAskView.setText(String.format(Locale.getDefault(), "%.8f", pair.getBittrexAsk()));
        holder.bittrexAskQuantityView.setText(String.format(Locale.getDefault(), "%.8f", pair.getBittrexAskQuantity()));
        holder.bittrexVolumeView.setText(String.format(Locale.getDefault(), "%.8f", pair.getBittrexVolume()));
        holder.percent.setText(String.valueOf(pair.getPercent()));
    }

    class PairsViewHolder extends RecyclerView.ViewHolder {
        final TextView pairNameView;
        final TextView bittrexAskView;
        final TextView bittrexAskQuantityView;
        final TextView bittrextBidView;
        final TextView bittrextBidQuantityView;
        final TextView bittrexVolumeView;
        final TextView livecoinAskView;
        final TextView livecoinAskQuantityView;
        final TextView livecoinBidView;
        final TextView livecoinBidQuantityView;
        final TextView livecoinVolumeView;
        final TextView percent;

        PairsViewHolder(@NonNull View itemView) {
            super(itemView);
            pairNameView = itemView.findViewById(R.id.pair_name);
            bittrexAskView = itemView.findViewById(R.id.bittrex_ask);
            bittrexAskQuantityView = itemView.findViewById(R.id.bittrex_ask_quantity);
            bittrextBidView = itemView.findViewById(R.id.bittrex_bid);
            bittrextBidQuantityView = itemView.findViewById(R.id.bittrex_bid_quantity);
            bittrexVolumeView = itemView.findViewById(R.id.bittrex_volume);
            livecoinAskView = itemView.findViewById(R.id.livecoin_ask);
            livecoinAskQuantityView = itemView.findViewById(R.id.livecoin_ask_quantity);
            livecoinBidView = itemView.findViewById(R.id.livecoin_bid);
            livecoinBidQuantityView = itemView.findViewById(R.id.livecoin_bid_quantity);
            livecoinVolumeView = itemView.findViewById(R.id.livecoin_volume);
            percent = itemView.findViewById(R.id.pair_percent);
        }
    }
}

File: app/src/main/java/com/chess/cryptobot/view/adapter/PairsAdapter.kt
package com.chess.cryptobot.view.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.chess.cryptobot.R
import com.chess.cryptobot.content.ContextHolder
import com.chess.cryptobot.market.Market
import com.chess.cryptobot.model.Pair
import com.chess.cryptobot.view.adapter.PairsAdapter.PairsViewHolder
import java.util.*

class PairsAdapter(pairsHolder: ContextHolder) : RecyclerViewAdapter<PairsViewHolder>(pairsHolder) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PairsViewHolder {
        val context = parent.context
        val layoutInflater = LayoutInflater.from(context)
        return PairsViewHolder(layoutInflater.inflate(R.layout.trading_pair_line, parent, false))
    }

    override fun onBindViewHolder(holder: PairsViewHolder, position: Int) {
        val pair = getItemByPosition(position) as Pair
        holder.pairNameView.text = pair.name
        holder.binanceBidView.text = String.format(Locale.getDefault(), "%.8f", pair.bidMap[Market.BINANCE_MARKET])
        holder.binanceBidQuantityView.text = String.format(Locale.getDefault(), "%.8f", pair.bidQuantityMap[Market.BINANCE_MARKET])
        holder.binanceAskView.text = String.format(Locale.getDefault(), "%.8f", pair.askMap[Market.BINANCE_MARKET])
        holder.binanceAskQuantityView.text = String.format(Locale.getDefault(), "%.8f", pair.askQuantityMap[Market.BINANCE_MARKET])
        holder.bittrexBidView.text = String.format(Locale.getDefault(), "%.8f", pair.bidMap[Market.BITTREX_MARKET])
        holder.bittrexBidQuantityView.text = String.format(Locale.getDefault(), "%.8f", pair.bidQuantityMap[Market.BITTREX_MARKET])
        holder.bittrexAskView.text = String.format(Locale.getDefault(), "%.8f", pair.askMap[Market.BITTREX_MARKET])
        holder.bittrexAskQuantityView.text = String.format(Locale.getDefault(), "%.8f", pair.askQuantityMap[Market.BITTREX_MARKET])
        holder.poloniexBidView.text = String.format(Locale.getDefault(), "%.8f", pair.bidMap[Market.POLONIEX_MARKET])
        holder.poloniexBidQuantityView.text = String.format(Locale.getDefault(), "%.8f", pair.bidQuantityMap[Market.POLONIEX_MARKET])
        holder.poloniexAskView.text = String.format(Locale.getDefault(), "%.8f", pair.askMap[Market.POLONIEX_MARKET])
        holder.poloniexAskQuantityView.text = String.format(Locale.getDefault(), "%.8f", pair.askQuantityMap[Market.POLONIEX_MARKET])
        holder.percent.text = pair.percent.toString()
    }

    class PairsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val pairNameView: TextView = itemView.findViewById(R.id.pair_name)
        val bittrexAskView: TextView = itemView.findViewById(R.id.bittrex_ask)
        val bittrexAskQuantityView: TextView = itemView.findViewById(R.id.bittrex_ask_quantity)
        val bittrexBidView: TextView = itemView.findViewById(R.id.bittrex_bid)
        val bittrexBidQuantityView: TextView = itemView.findViewById(R.id.bittrex_bid_quantity)
        val binanceAskView: TextView = itemView.findViewById(R.id.binance_ask)
        val binanceAskQuantityView: TextView = itemView.findViewById(R.id.binance_ask_quantity)
        val binanceBidView: TextView = itemView.findViewById(R.id.binance_bid)
        val binanceBidQuantityView: TextView = itemView.findViewById(R.id.binance_bid_quantity)
        val poloniexAskView: TextView = itemView.findViewById(R.id.poloniex_ask)
        val poloniexAskQuantityView: TextView = itemView.findViewById(R.id.poloniex_ask_quantity)
        val poloniexBidView: TextView = itemView.findViewById(R.id.poloniex_bid)
        val poloniexBidQuantityView: TextView = itemView.findViewById(R.id.poloniex_bid_quantity)
        val percent: TextView = itemView.findViewById(R.id.pair_percent)
    }
}